Lib/test/test_unittest/testmock/testsealable.py

```diff
@@ -200,6 +200,9 @@ def ban(self):
                 self.assertIsInstance(foo.Baz.baz, mock.NonCallableMagicMock)
                 self.assertIsInstance(foo.Baz.ban, mock.MagicMock)
 
+                # see gh-91803
+                self.assertIsInstance(foo.bar2(), mock.Mock)
```

Comment:
Thanks for fixing the conflicts. I guess its good to have `MagicMock` explicitly since child_klass is `MagicMock` though MagicMock passes isinstance for both `Mock` and `MagicMock`.

```
>>> isinstance(unittest.mock.MagicMock(), unittest.mock.Mock)
True
>>> isinstance(unittest.mock.MagicMock(), unittest.mock.MagicMock)
True
```

```suggestion
                self.assertIsInstance(foo.bar2(), mock.MagicMock)
```